module/zfs/zfs_vnops.c

```diff
@@ -829,6 +829,22 @@ zfs_write(struct inode *ip, uio_t *uio, int ioflag, cred_t *cr)
 			uio->uio_fault_disable = B_FALSE;
 			if (error == EFAULT) {
 				dmu_tx_commit(tx);
+				/*
+				/ This is the only special case in the loop
+				/ where we "continue" to the next iteration,
+				/ and if we execute this code, we did a partial copy.
+				/ Account for bytes written, otherwise we end up
+				/ with invalid sizes/offsets at the next iteration.
+				/ Also, this needs to be done before the following
+				/ uio_prefault. Otherwise, in case we're writing
+				/ the last chunk, we we try to prefault n bytes
+				/ (<max_blksz), but n could be bigger than the
+				/ data itself, generate an EFAULT, and make
+				/ the loop break early without completing the
+				/ last chunk.
+				*/
```

Comment:
nit: the detailed comment is welcome, but perhaps it could be a little more concise.  Also please use the existing comment style found elsewhere in this function.  You can run `make checkstyle`  locally to verify the formatting.

Reply:
Fixed all "make checkstyle" issues, and made the comment shorter.
Let me know if it's ok.